### PR TITLE
Ping the websocket server in order to reconnect when the connection closed

### PR DESCRIPTION
**Pull request by noramehesz**

### CODE REVIEW

File: lib/resources/datav2/websocket.ts [hunk 1]
```diff
@@ -213,6 +216,15 @@ export abstract class AlpacaWebsocket
     }
   }
 
+  ping(): void {
+    this.conn.ping(this.msgpack.encode(this.session.pingCounter));
+    this.session.pingCounter++;
+    this.session.pongTimeout = setTimeout(() => {
+      this.log("connection may closed, terminating...");
```
Comment (gnvk) [inline lib/resources/datav2/websocket.ts:223]: [minor] I prefer more explicit logs, e.g. "no pong received from server, terminating..."

File: test/support/mock-streaming.js [hunk 2]
```diff
@@ -196,7 +199,7 @@ class StreamingWsMock {
         dailyBars: this.subscriptions.dailyBars,
         statuses: this.subscriptions.statuses,
         lulds: this.subscriptions.lulds,
-        cancelErrors: this.subscriptions.trades,  // Subscribed automatically.
+        cancelErrors: this.subscriptions.trades, // Subscribed automatically.
```
Comment (gnvk) [inline test/support/mock-streaming.js:202]: [minor][non-related] please also fix the double // in the next line

File: lib/resources/datav2/websocket.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import events from "events";
 import WebSocket from "ws";
 import { MessagePack } from "msgpack5";
 import msgpack5 from "msgpack5";
+import { timeStamp } from "console";
```
Comment (gnvk) [inline lib/resources/datav2/websocket.ts:5]: [minor] Can you please tell me where/how this is used?

File: lib/resources/datav2/websocket.ts [hunk 4]
```diff
@@ -76,8 +77,10 @@ interface WebsocketSession {
   backoffIncrement: number;
   url: string;
   currentState: STATE;
-  pingTimeout?: NodeJS.Timeout;
-  pingTimeoutThreshold: number;
+  pongTimeout?: NodeJS.Timeout;
+  pingInterval?: NodeJS.Timer;
+  pongWait: number;
+  pingCounter: number;
```
Comment (gnvk) [inline lib/resources/datav2/websocket.ts:83]: Why is this needed?

File: lib/resources/datav2/websocket.ts (hunk 3 above)
Comment (noramehesz) [reply lib/resources/datav2/websocket.ts:5]: Forgot to delete

File: lib/resources/datav2/websocket.ts (hunk 4 above)
Comment (noramehesz) [reply lib/resources/datav2/websocket.ts:83]: According to the standard sending payload in a ping message is optional. I almost removed it but the go's websocket library sends incrementing number and I decided doing the same. Should I remove?

Comment (gnvk) [reply lib/resources/datav2/websocket.ts:83]: I think it's fine to remove it, the server is doing nothing with it.

Comment (noramehesz) [reply lib/resources/datav2/websocket.ts]: Removed

**Review by gnvk — APPROVED** [review]
LGTM

**Review by Mark-Pintye — APPROVED** [review]
